examples/vault-ec2-auth/README.md

```diff
@@ -0,0 +1,171 @@
+# Vault authentication using EC2 metadata example
+
+This example shows how to use the metadata from an EC2 instance to authenticate
+to a [vault cluster][vault_cluster].
+
+Vault provides multiple [auth methods][auth_methods] such as Username & Password, GitHub
+etc. Among those methods you will find AWS. The way it works is that it
+understands [AWS][aws_auth] as a trusted third party, and relies on AWS itself for affirming
+if an authentication source such as an EC2 Instance or other resources like a
+Lambda Function are legitimate sources or not. Basically, if AWS trusts the
+origin, then so do we.
+
+There are currently two ways an AWS resource can authenticatate: `ec2` and `iam`. In
+this example, we will explore the first option.
+
+**Note**: To keep this example as simple to deploy and test as possible and because we are
+focusing on authentication, it deploys the Vault cluster into your default VPC and default subnets,
+ all of which are publicly accessible. This is OK for learning and experimenting, but for
+production usage, we strongly recommend deploying the Vault cluster into the private subnets
+of a custom VPC.
+
+## Running this example
+You will need to create an [Amazon Machine Image (AMI)][ami] that has Vault and Consul installed,
+which you can do using the [vault-consul-ami example][vault_consul_ami]). Each of the servers
+in this example, including the instance that is authenticating to Vault has [Dnsmasq][dnsmasq]
+installed (via the [install-dnsmasq module][dnsmasq_module]) which allows it to use the Consul
+server cluster for service discovery and thereby access Vault via DNS using the domain name
+`vault.service.consul`.
+
+### Quick start
+
+1. `git clone` this repo to your computer.
+1. Build a Vault and Consul AMI. See the [vault-consul-ami example][vault_consul_ami] documentation for
+   instructions. Make sure to note down the ID of the AMI.
+1. Install [Terraform](https://www.terraform.io/).
+1. Open `vars.tf`, set the environment variables specified at the top of the file, and fill in any other variables that
+   don't have a default. If you built a custom AMI, put the AMI ID into the `ami_id` variable. Otherwise, one of our
+   public example AMIs will be used by default. These AMIs are great for learning/experimenting, but are NOT
+   recommended for production use.
+1. Run `terraform init`.
+1. Run `terraform apply`.
+1. Run the [vault-examples-helper.sh script](https://github.com/hashicorp/terraform-aws-vault/tree/master/examples/vault-examples-helper/vault-examples-helper.sh) to
```

Comment:
Probably for this example, the user should curl the server to see that it was able to read the secret from vault, right?